### PR TITLE
Use `jnr-posix-api` plugin

### PR DESCRIPTION
**Pull request by sarah-witt**
### What does this PR do?
Adds jnr-posix-api as a dependency to the plugin and bump the minimum required jenkins version

### Description of the Change
Jenkins core removed [JNR](https://github.com/jenkinsci/jenkins/pull/6323), and the java dogstatsd client relies on this library. To add the plugin, we need to bump the minimum required jenkins version.

Addresses https://github.com/jenkinsci/datadog-plugin/issues/285

### Additional Notes

The name of the built-in node changed in more recent versions of jenkins, so we will have to update our tests in order to test them in our jenkinsfile: https://www.jenkins.io/changelog-stable/#v2.319.1

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

**Review by ofek — APPROVED** [review]
LGTM

File: src/test/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListenerIT.java [hunk 1]
```diff
@@ -147,7 +147,7 @@ public void testTraces() throws Exception {
         assertEquals("success", meta.get(CITags.STATUS));
         assertNotNull(meta.get(CITags.NODE_NAME));
         assertNotNull(meta.get(CITags.NODE_LABELS));
-        assertNull(meta.get(CITags._DD_HOSTNAME));
```
Comment (drodriguezhdez) [inline src/test/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListenerIT.java:150]: @sarah-witt I think we cannot change this. The hostname must be "null".

Comment (drodriguezhdez) [reply src/test/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListenerIT.java:150]: It's been resolved in this PR: https://github.com/jenkinsci/datadog-plugin/pull/288

**Comment by basil** [comment]
BTW `jnr-posix-api` is part of the BOM as of https://github.com/jenkinsci/bom/releases/tag/1556.vfc6a_f216e3c6, so once you upgrade to that BOM version (which also requires upgrading to a baseline of 2.319.1 or later) you can remove the `<version>` from `jnr-posix-api` and get the version from the BOM instead, which should simplify maintenance.

**Comment by sarah-witt** [comment]
Thanks for the info @basil! We are going to bump the baseline with this [PR](https://github.com/jenkinsci/datadog-plugin/pull/296/files) so I will make note to do this after that PR is merged.

**Pull request merged**